### ISSUE TITLE
darknet_video.py off by 1 frame

### ISSUE THREAD

**Issue by mazatov** [AlexeyAB/darknet#6706]
Running darknet_video.py on the video makes a video where the detections and the video are off by one frame. 

The amount of frames it's off by seems to depend on `max_size` of the `darknet_image_queue`

**Comment by mazatov** [comment]
Also, not clear why `max_size =1` for the queues. Larger `max_size` would speed things up. Seems like the author suspected that to be a problem.

**Comment by mazatov** [comment]
Update. The darknet_image is overwritten in video_capture every time it's being created because it sits in the same memory so effectively the darknet_image_queue is not working properly

Proposed fix [here](https://github.com/AlexeyAB/darknet/pull/6717):

First time doing PR here @AlexeyAB , so let me know if I need to do more. Seems like `darknet detector demo` has the same error though so possibly it's in C++ code too but I don't know enough to make changes there.

**Comment by Tacokeet** [comment]
Hey @mazatov, I also found this problem, thanks for the solution. The delay within the `darknet detector demo` can be fixed by adding the flag `-avgframes 1` found here #5806.

**Comment by mazatov** [comment]
Thanks didn't know about this `arg` option. 

Still, it's a bit of hackish fix as it assumes that `cv2.videoreader` is going to work faster than `darknet.detect_image` and the frame in the queue will always be overwritten by the next frame. While it's true for most computer hardware, I can imagine a situation where `detect_image` outperformed `cv2.videoreader` and you would suddenly not have that 1 shift in frames.

**Comment by Tacokeet** [comment]
Okay I just tried it out on my computer.
It fixes the 1 frame problem but there is memory problem. It keeps ramping up the memory usage until it crashes.

**Comment by mazatov** [comment]
Good point, if you have a long video and GPU can't keep up with the loading it would happen. I should add maxsize to the queues so they don't load too much into memory

**Comment by mazatov** [comment]
@Tacokeet , I was still getting a memory problem so I worked on it a bit more. I did a hacky fix to the issue, but it works. You can check the updated PR.

**Comment by Tacokeet** [comment]
Thanks for the update! Seems you forgot to push the correct version because 'darknet_images' is not defined. But after adding it seems to run nice. I can now finish a ~9 min video without any memory issues.

**Comment by mazatov** [comment]
Oh, right! Thanks for pointing it out!
